pubsub/tests/unit/pubsub_v1/publisher/batch/test_thread.py

```diff
@@ -314,22 +316,28 @@ def test_publish():
     assert len(batch.messages) == 3
     assert batch._futures == futures
 
-    # The size should have been incremented by the sum of the size of the
-    # messages.
-    expected_size = sum([message_pb.ByteSize() for message_pb in messages])
-    assert batch.size == expected_size
+    # The size should have been incremented by the sum of the size
+    # contributions of each message to the PublishRequest.
+    base_request_size = types.PublishRequest(topic="topic_foo").ByteSize()
+    msg_size_overheads = [
+        types.PublishRequest(messages=[msg]).ByteSize() for msg in messages
+    ]
+
+    expected_request_size = base_request_size + sum(msg_size_overheads)
```

Comment:
Just do the comprehension in the sum and cut out the `msg_size_overheads` variable.

Reply:
I personally find it beneficial to give some constructs a concrete name, can improve readability of a test.\

Unless you find the current version unpleasantly verbose?

Reply:
I agree, lifting constructs can greatly improve readability, reduce the rightward march, and makes pdb based debugging _way_ easier. I'm also a big proponent of dataflow driven programming, though, and balancing those forces can be tricky sometimes.